src/app/containers/ArticleTimestamp/index.jsx

```diff
@@ -59,11 +64,28 @@ const ArticleTimestamp = ({
 
   const timeDifferenceMinutes = (lastPublished - firstPublished) / 1000 / 60;
 
+  const publishedAndUpdatedOnSameDay = isSameDay(firstPublished, lastPublished);
+
+  const publishedToday = isToday(firstPublished);
+
+  const updatedLessThanTenHoursAgo = isLastRelative(lastPublished);
+
+  const publishedAndUpdatedToday =
+    publishedAndUpdatedOnSameDay && publishedToday;
+
+  const publishedAndUpdatedOnSameDayOverTenHoursAgo =
+    publishedAndUpdatedOnSameDay && !updatedLessThanTenHoursAgo;
+
+  const showLastUpdatedTimestamp =
+    (publishedAndUpdatedToday ||
+      !publishedAndUpdatedOnSameDayOverTenHoursAgo) &&
+    timeDifferenceMinutes > minutesTolerance;
```

Comment:
I am tempted to say - can we move this logic into a separate file, and add a chunk of tests, just to make sure it is bullet proof.  I know we have a snapshot test here, but I would prefer a unit test which will return true/false based on the firstPublished & lastPublished values passed to it :/

Reply:
I think this is a great shout and will build confidence in the change. Could create a new folder with these in along similar lines to `timeFormatTests`? Not sure what the precedent is.

Reply:
I find the negated `!publishedAndUpdatedOnSameDayOverTenHoursAgo` a little hard to follow (maybe because of the negated `!updatedLessThanTenHoursAgo` 'within' it?). Was playing around with some other ways of writing it, what do you think about something like this? It's a little more verbose I admit.

```javascript
const hasBeenUpdated = timeDifferenceMinutes > minutesTolerance;

const showLastUpdatedTimestamp =
  hasBeenUpdated &&
  (publishedAndUpdatedToday ||
    updatedLessThanTenHoursAgo ||
    !publishedAndUpdatedOnSameDay);
```

Reply:
@karinathomas wrote:
> I know we have a snapshot test here, but I would prefer a unit test

Agree with this, I was going to say the fact that we only had to update a snapshot to account for the changed behaviour suggests to me some more explicit assertion tests might be needed.

Reply:
Yea these suggestions and comments are all great thanks guys - learning day tomorrow so I likely won't pick back up until Monday but will nudge you when updated

Reply:
Reconfigured and renamed the variables in the latest commit. No more double negatives, or a flipping of a 'hidden' scenario

Reply:
Reconfigured and renamed the variables in the latest commit. No more double negatives, or a flipping of a 'hidden' scenario